### PR TITLE
R 4.1, new cluster, local win tests

### PR DESCRIPTION
**Pull request by weshinsley**
I have written an essay here...!
https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-2425

### CODE REVIEW

File: R/client.R [hunk 1]
```diff
@@ -409,7 +409,7 @@ client_parse_r_versions <- function(txt) {
 
 client_parse_headnodes <- function(txt) {
   dat <- strsplit(txt, "\n")[[1]]
-  stopifnot(all(grepl("^fi--", dat)))
+  stopifnot(all(grepl("^fi--|wpia-", dat)))
```
Comment (richfitz) [inline R/client.R:412]:
```suggestion
  stopifnot(all(grepl("^(fi--|wpia-)", dat)))
```

is the wpia prefix a single prefix? oh dear

File: R/config.R [hunk 2]
```diff
@@ -392,7 +393,8 @@ cluster_name <- function(name) {
     if (!(name %in% valid_clusters())) {
       alias <- list(
         "fi--dideclusthn" = c("small", "little", "dide", "ide", "dideclusthn"),
-        "fi--didemrchnb" = c("big", "mrc", "didemrchnb"))
+        "fi--didemrchnb" = c("big", "mrc", "didemrchnb"),
+        "wpia-hpc-hn" = c("covid"))
```
Comment (richfitz) [inline R/config.R:397]: No `c()` here

File: R/config.R [hunk 3]
```diff
@@ -405,7 +407,8 @@ cluster_name <- function(name) {
 valid_templates <- function() {
   list("fi--dideclusthn" = c("GeneralNodes", "8Core", "Training"),
        "fi--didemrchnb" = c("GeneralNodes", "12Core", "12and16Core", "16Core",
-                            "20Core", "24Core", "32Core"))
+                            "20Core", "24Core", "32Core"),
+       "wpia-hpc-hn" = c("AllNodes"))
```
Comment (richfitz) [inline R/config.R:411]: No `c()` here

File: R/config.R [hunk 4]
```diff
@@ -418,7 +421,9 @@ check_resources <- function(cluster, template, cores, wholenode, parallel) {
       stop("Cannot specify both wholenode and cores")
     }
     assert_scalar_integer(cores)
-    max_cores <- if (cluster == "fi--didemrchnb") 64 else 24
+    max_cores <- 24
+    if (cluster == "fi--didemrchnb") max_cores <- 64
+    else if (cluster == "wpia-hpc-hn") max_cores <- 32
```
Comment (richfitz) [inline R/config.R:426]: can you move this into a free function?

```r
didehpc_max_cores <- function(cluster) {
 switch(cluster,
        "fi--dideclusthn" = 24
        "fi--didemrchnb" = 64,
        "wpia-hpc-hn" = 32,
        stop(sprintf("Invalid cluster '%s'", cluster))
}
```

and use as

```r
max_cores <- didehpc_max_cores(cluster)
```

or put the check in there too?

File: R/queue_library.R [hunk 5]
```diff
@@ -63,8 +62,10 @@ queue_library <- R6::R6Class(
 queue_template <- function(cluster) {
```
Comment (richfitz) [inline R/queue_library.R:62]: Convert to switch, add default error case

File: tests/testthat/test-mounts.R [hunk 6]
```diff
@@ -98,12 +99,13 @@ test_that("detect_mount_windows tries different methods in turn", {
   mockery::stub(detect_mount_windows, "wmic_call", mock_wmic_call)
 
   expect_equal(detect_mount_windows(), res$result)
-
+  win_dir <- Sys.getenv("windir")
+  if (win_dir == "") win_dir <- "C:\\Windows"
```
Comment (richfitz) [inline tests/testthat/test-mounts.R:103]:
```suggestion
  win_dir <- Sys.getenv("windir", "C:\\Windows")
```

File: tests/testthat/test-mounts.R [hunk 7]
```diff
@@ -115,11 +117,13 @@ test_that("detect_mount_windows errors if no method found", {
     detect_mount_windows(),
     "Could not determine windows mounts using wmic.+some error")
   mockery::expect_called(mock_wmic_call, 3)
+  win_dir <- Sys.getenv("windir")
+  if (win_dir == "") win_dir <- "C:\\Windows"
```
Comment (richfitz) [inline tests/testthat/test-mounts.R:121]:
```suggestion
  win_dir <- Sys.getenv("windir", "C:\\Windows")
```

**Review by richfitz — CHANGES_REQUESTED** [review]
A few small things in here

File: R/client.R (hunk 1 above)
Comment (weshinsley) [reply R/client.R:412]: "wpia-" is the new "fi--" - apparently it is funding specific.

Comment (richfitz) [reply R/client.R]: I understand the fi/wpia part, but surprised at the hyphen number is all

**Review by richfitz — APPROVED** [review]

**Pull request merged**